lib/stub_alloc_pages.c

```diff
@@ -58,7 +58,9 @@ caml_alloc_pages(value did_gc, value n_pages)
   void* block = _xmalloc(len, PAGE_SIZE);
   if (block == NULL) {
 #elif _WIN32
-  void *block = _aligned_malloc(len, PAGE_SIZE);
+  /* NB we can't use _aligned_malloc because we can't get OCaml to
+     finalize with _aligned_free. Regular free() will not work. */
+  void *block = NULL;
```

Comment:
You can get this to work by setting the allocation mask to `CAML_BA_EXTERNAL` and then calling the right free function in a finalizer.  Thing is, we don't take advantage of the page alignment on Windows at any point...